lib/cluster-providers/generic-cluster-provider.ts

```diff
@@ -189,10 +191,10 @@ export class GenericClusterProvider implements ClusterProvider {
             "Mixing managed and autoscaling node groups is not supported. Please file a request on GitHub to add this support if needed.");
     }
 
-    /** 
-     * @override 
+    /**
+     * @override
      */
-    createCluster(scope: Construct, vpc: ec2.IVpc): ClusterInfo {
+    createCluster(scope: Construct, vpc: IVpc, secretsEncryptionKey: IKey): ClusterInfo {
```

Comment:
while VPC is a mandatory parameter, secretsEncryptionKey seems to be optional since the default `KmsKeyProvider` (no-arg constructor) returns undefined. Do you see any issue making it an optional parameter here, e.g. `secretEncryptionKey?`?

Reply:
> while VPC is a mandatory parameter, secretsEncryptionKey seems to be optional since the default `KmsKeyProvider` (no-arg constructor) returns undefined. Do you see any issue making it an optional parameter here, e.g. `secretEncryptionKey?`?

Should we make the KMS Key Provider optional and not default?

Reply:
You are right, it is consistent.